### ISSUE TITLE
Feature request: Change disc for multi-disc games

### ISSUE THREAD

**Issue by prataczak** [libretro/dolphin#36]
There's some great multi disc titles for Gamecube, but there is no way to change discs with the RetroArch core.

Standalone version of Dolphin does this. It would be a great addition to the libretro core.

**Comment by Papermanzero** [comment]
M3U approach like in beetle psx would be great

**Comment by klepp0906** [comment]
+1
m3u is a distant but welcome last place.  Id have to restructure my parser for steam but beggars cant be choosers.  Actually now that I think of it, im not sure m3u would even work.  Most gamecube games are single isos thus have no cue sheet to point an m3u to.  IIRC how m3u functions anyhow.  been a minute :)

a native switching capability via core options is preferable but I'm wholly understanding of the path of least resistance. 

what i do know is for my  needs and preferences, retroarch is superior to any alternative and I use a RA core in any and EVERY instance I can.  

Not having the ability to use multi disc titles on dolphin makes that an impossibility since some of its best games are multi disc.

Fortunately this is a problem that all but disappeared post ps2 generation i believe.

**Comment by prataczak** [comment]
Well I use RocketLauncher and Launchbox/HyperSpin, so for me, the m3u thing is ideal, since RocketLauncher's Pause automates m3u disc swapping.

If I can get a dev to show some interest in this with m3u support, I'd personally drop at least $50 on it. Otherwise, I'd put up $15.

**Comment by anothername99** [comment]
You don't need a cue sheet to use m3u. All you need is to point to which files to load. So for GameCube you would do this, if m3u support got implemented:

Resident Evil 0 Disc 1.iso
Resident Evil 0 Disc 2.iso.

**Comment by klepp0906** [comment]
ive created the m3u's in the hopes/anticipation that a developer will see and add this.   I went to use standalone dolphin in the interim, but it also doesnt have m3u support in.  The developer there says it is on his to-do list.

normally id default to stand-alone until it was added in RA core, however since i do all my emulation via steam link remotely, its ability to switch discs via the interface isnt much help, at least not the way I have it set up via steam shortcuts (to open games directly/bypassing the emulator interface). 

anyhow, i digress.

**Comment by i30817** [comment]
there is no reason for m3u to be limited to cues. Indeed, originally the format was for mp3 albums.

Basically it should be used for all kinds of 'index files' or files that are selfcontained.

I actually really dislike the .cmd workaround in certain systems for the lack of m3u, like pc-98 emulators. I understand it's because of lack of a parser for that core i guess.

**Comment by klepp0906** [comment]
yea, i was misinformed.  It does in fact index w/e you'd like in regard to disc images, providing support exists of course.

If anyone hears a peep about this being on the table - please keep me posted.   Id definitely try to chip in on a bounty for it if necessary.  I want to use retroarch exclusively but cores that dont support multiple discs is a huge hurdle to me doing so.

**Comment by mccorkled** [comment]
> 
> 
> yea, i was misinformed. It does in fact index w/e you'd like in regard to disc images, providing support exists of course.
> 
> If anyone hears a peep about this being on the table - please keep me posted. Id definitely try to chip in on a bounty for it if necessary. I want to use retroarch exclusively but cores that dont support multiple discs is a huge hurdle to me doing so.

I'm with you on this. I used RA way back when it had it's first GUI and I swore I would never use it again. Now I am using it for every system I can, but games with multidiscs. Looking forward to this implementation big time. I made a thread about it [here ](https://forums.dolphin-emu.org/Thread-multi-disc-support-with-ahk)and linked to some RocketLauncher threads asking for support as well.

**Comment by flyinghead** [comment]
This feature has been added. Please test.

**Comment by prataczak** [comment]
I updated the core, and I'm not seeing 'disc control' in the quick menu. Can you give more details on how it was implemented please? Is it the m3u style? Can I press the 'change disc' keys to change disc? I will gladly test it as soon as I understand how it works.

**Comment by flyinghead** [comment]
You should see the Disk Control entry in the Quick Menu.

Use Disk Image Append to open a new disc image.

![dolphin disc swap](https://user-images.githubusercontent.com/16261910/50366361-7e521100-0579-11e9-8610-66991367be1b.png)

**Comment by prataczak** [comment]
Ok, the dolphin core I'm using after updating is 1.7.5 - dolphin-emu a5bce7d. Perhaps your version will be available for download at a later time.

**Comment by flyinghead** [comment]
It looks like the core is not building on some platforms including Windows, which probably explains why you're not seeing any change

**Comment by prataczak** [comment]
Makes sense, I'm on Windows.

**Comment by flyinghead** [comment]
The Windows build finally went through so this feature should now be available on all platforms.

**Comment by prataczak** [comment]
I tested with Baten Kaitos - Eternal Wings and the Lost Ocean (USA) (Disc 1) and I tried the 'Disk Image Append' to load Baten Kaitos - Eternal Wings and the Lost Ocean (USA) (Disc 2) and RetroArch crashed and closed on me.

I'll play around with it some more later.

**Comment by prataczak** [comment]
Ok, I tested with the same game you used (Tiger Woods PGA Tour 2005 (USA) (Disc 2) ) and it immediately asked for Disc 1, and I was able to successfully append the disc to Disc 1.

I'm assuming my previous crash occurred because I tried to change the disc at a time when the game wasn't asking for a disc change. At any rate, it appears to be working, so I'll go ahead and close this issue.

Thanks a lot for adding this!
